cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -91,6 +91,25 @@ TYPED_TEST(TestStringKernels, AsciiLower) {
                    "[\"aaazzæÆ&\", null, \"\", \"bbb\"]");
 }
 
+TYPED_TEST(TestStringKernels, AsciiReverse) {
+  this->CheckUnary("ascii_reverse", "[]", this->type(), "[]");
+  this->CheckUnary("ascii_reverse", R"(["abcd", null, "", "bbb"])", this->type(),
+                   R"(["dcba", null, "", "bbb"])");
+
+  Datum input = ArrayFromJSON(this->type(), "[\"aAazZæÆ&\", null, \"\", \"bbb\"]");
+  ASSERT_NOT_OK(CallFunction("ascii_reverse", {input}));
```

Comment:
Please test the error return more precisely, instead of only checking that it's not ok.